### PR TITLE
Only search on ISearchReplaceable interface

### PR DESCRIPTION
**Pull request by Gagaro**
Based on https://github.com/collective/collective.searchandreplace/pull/3

### CODE REVIEW

File: collective/searchandreplace/profiles/dexterity/types/File.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0"?>
+<object name="File" meta_type="Dexterity FTI" i18n:domain="plone"
+   xmlns:i18n="http://xml.zope.org/namespaces/i18n">
+ <property name="behaviors" purge="true">
```
Comment (mauritsvanrees) [inline collective/searchandreplace/profiles/dexterity/types/File.xml]: This should be `purge="false"` like the others.

File: collective/searchandreplace/searchreplaceutility.py [hunk 2]
```diff
@@ -55,7 +55,7 @@ def searchObjects(self, context, find, **kwargs):
         if isDefaultPage(container, context) and ssf:
             query['query'] = '/'.join(container.getPhysicalPath())
         catalog = getToolByName(context, 'portal_catalog')
-        brains = catalog(path=query)
+        brains = catalog(path=query, object_provides='collective.searchandreplace.interfaces.ISearchReplaceable')
```
Comment (mauritsvanrees) [inline collective/searchandreplace/searchreplaceutility.py]: Nit picking: I prefer a line length limit of 80.

**Comment by mauritsvanrees** [comment]
When I go to a content item that does not have the required interface, I still see the Search/Replace tab. When I click on it, I get a 404 error page.  The tab should get some extra condition so it is only shown when it is available.
Alternatively, I guess registering this form for every page is fine too: for example when you make only pages ISearchReplaceable and folders and the Plone Site not, then how are you going to replace text in more than one page?

BTW, to test this, I commented out the zcml that makes Documents ISearchReplaceable and was surprised that text was still replaced in them.  Then I realised that the catalog had no way of knowing that documents no longer provided this interface: after reindexing this index, all was well. For clarity: no reindexing is needed in our code, it is just something to remember when you are testing.

**Comment by Gagaro** [comment]
Yes maybe we should change which content types have the interface. The behaviour of the module is being changed though, maybe we should make a new major version for these changes.

**Comment by Gagaro** [comment]
I changed the view so it can be used anywhere. Only the contents with the interface are searched for. After testing it, I think it's actually a better behaviour than currently.

**Comment by mauritsvanrees** [comment]
Thanks for your work.
I have updated the documentation and added a warning. Can you check?

**Comment by Gagaro** [comment]
Looks good to me, nice catch on the check after the catalog query. Let me just add the french translation and this should be good to go.

**Comment by Gagaro** [comment]
Done

**Comment by mauritsvanrees** [comment]
I have released 6.0. Thanks!

**Pull request merged**